zerver/actions/create_realm.py

```diff
@@ -161,6 +161,8 @@ def do_create_realm(
     if org_type is not None:
         kwargs["org_type"] = org_type
     if enable_spectator_access is not None:
+        if enable_spectator_access and plan_type == Realm.PLAN_TYPE_LIMITED:
+            enable_spectator_access = False
```

Comment:
We pass `enable_spectator_access` and `plan_type` to `do_create_realm` only when calling it from `populate_db`. Also, for now I have kept it to set it as False in that case, rather than raising error.

Reply:
Maybe better would be this?

`assert !enable_spectator_access or plan_type != Realm.PLAN_TYPE_LIMITED`

With a comment that we're enforcing the invariant that the LIMITED plan type isn't allow to have spectators enabled.